clang/include/clang/AST/Expr.h

```diff
@@ -3518,6 +3523,13 @@ class CompoundLiteralExpr : public Expr {
     TInfoAndScope.setPointer(tinfo);
   }
 
+  bool hasStaticStorage() const { return isFileScope() && isGLValue(); }
+  APValue *getOrCreateStaticValue(ASTContext &Ctx) const;
```

Comment:
Suggestion: consistently return a reference or a pointer in both functions? (either of the two is fine, it's just a little confusing to see both)